src/Symfony/Component/DomCrawler/AbstractUriElement.php

```diff
@@ -40,15 +40,17 @@
      *
      * @throws \InvalidArgumentException if the node is not a link
      */
-    public function __construct(\DOMElement $node, string $currentUri, ?string $method = 'GET')
+    public function __construct(\DOMElement $node, string $currentUri = null, ?string $method = 'GET')
     {
-        if (!\in_array(strtolower(substr($currentUri, 0, 4)), array('http', 'file'))) {
-            throw new \InvalidArgumentException(sprintf('Current URI must be an absolute URL ("%s").', $currentUri));
-        }
-
         $this->setNode($node);
         $this->method = $method ? strtoupper($method) : null;
         $this->currentUri = $currentUri;
+
+        $elementUriIsRelative = null === parse_url(trim($this->getRawUri()), PHP_URL_SCHEME);
+        $baseUriIsAbsolute = \in_array(strtolower(substr($this->currentUri, 0, 4)), array('http', 'file'));
```

Comment:
Should we use `strpos()` here instead?

Reply:
It's the same code as in the older branches, so we know it works and I'd prefer to not change it 😅